### ISSUE TITLE
Monte-Carlo Dropout not working on M1 (without seed)

### ISSUE THREAD

**Issue by chaudharyachint08** [keras-team/keras#16720]
**System information**
- Apple M1 chipset running "macOS Monterey 12.4"
- Tensorflow installed from binary (version 2.9.1)
- TensorFlow version (use command below): 2.9.1
- Python version: 3.9.13
- GPU model and memory: Apple M1 and metal GPU
- Miniconda as package manager as described in
https://developer.apple.com/metal/tensorflow-plugin/
Below are the exact installation commands
<pre>
conda install -c apple tensorflow-deps --yes
pip install --upgrade tensorflow-macos tensorflow-metal
</pre>

**Bug description**.
While using the Monte-Carlo Dropout (Dropout layer with training=True), Model is giving same output every-time (same connections are being dropped on each different call). 

**Describe the current behaviour**
I have provided a code for reproducing the issue, below is current output which I am getting from 3 different calls to MC-dropout layer
<pre>
[[-0.     -0.9094 -0.     -1.6735 -0.    ]]
[[-0.     -0.9094 -0.     -1.6735 -0.    ]]
[[-0.     -0.9094 -0.     -1.6735 -0.    ]]
[[-0.     -0.9094 -0.     -1.6735 -0.    ]]
</pre>

**Expected output**
Upon passing the same data multiple times (on each call) we should see different connections to be dropped, and hence get different output (which is the case when I am running the same code & TF 2.9.1 on Google Colab, output mentioned below)
<pre>
[[-1.1801 -0.     -1.1067 -1.6735 -0.7348]]
[[-0.     -0.9094 -0.     -1.6735 -0.    ]]
[[-0.     -0.     -1.1067 -0.     -0.    ]]
[[-0.     -0.     -0.     -1.6735 -0.    ]]
</pre>

**Standalone code to reproduce the issue**
<pre>
import numpy as np
from tensorflow import keras
from tensorflow.keras import layers

class MonteCarloDropout(layers.Dropout):
    def call(self, inputs):
        return super().call(inputs, training=True)

n_samples = 5
ip_dim, op_dim = 4, 1

SEED_VAL = 0
np.random.seed(SEED_VAL)
# keras.utils.set_random_seed(SEED_VAL)

X = np.random.random((n_samples, ip_dim))

fc_layer = layers.Dense(op_dim, input_shape=(ip_dim,))
mc_layer = MonteCarloDropout(rate=0.5)

for val in range(4):
	print( mc_layer(fc_layer(X)).numpy().T.round(4) )
</pre>

Also, find the link to the Jupyter [notebook](https://colab.research.google.com/drive/1rjpXMnhsOJ8nYVKNb3IbEcvXeAQ0VO1y?usp=sharing) on Google colab

**Additional note**
<B>Line 17</B> on Jupyter notebook, if uncommented solves the issue temporarily. But setting a random seed makes entire procedure deterministic which should not be the case

**Comment by tilakrayal** [comment]
@gadagashwini,
I was able to reproduce the issue on tensorflow v2.8, v2.9 and nightly. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/61b5b718e9e07f032bbc509d8bc865c9/16720.ipynb).

**Comment by haifeng-jin** [comment]
@chaudharyachint08 @tilakrayal I am not able to reproduce the issue with the either of the 2 notebooks.
It always give different output everytime, which is expected.

Please reopen if you still can reproduce it.

**Comment by haifeng-jin** [comment]
@chaudharyachint08 I do not have the environment to reproduce this bug. So is it only reproducable on M1?

**Comment by aliakbars** [comment]
Hi @haifeng-jin, the same thing happened to me on my M1 Macbook as well.

**Comment by chaudharyachint08** [comment]
Yes @haifeng-jin , I tried reproducing the bug on Nvidia GPU & AMD CPU based TensorFlow installs too, only comes up on M1 machine
